app/src/marketplace/modules/dataUnion/services.js

```diff
@@ -35,9 +35,15 @@ export const createJoinPartStream = async (account: Address, productId: ProductI
 
     let stream
     try {
-        stream = await postStream(newStream)
+        try {
+            stream = await getStream(newStream.id)
```

Comment:
I wonder if there is a code or something that we can detect if the creation fails because of duplicate? Maybe getting the stream should be the backup, since I guess that the stream creation shouldn' fail that often... that said, maybe not worth fine tuning this since this feature is going away.

Reply:
Yeah, thought the same. Deploying DUs is rare so the extra `GET` call should not matter that much. Of course it would be nice to get good behaving responses from the backend.